src/python/pants/util/meta.py

```diff
@@ -137,4 +137,8 @@ def new_setattr(self, key: str, value: Any) -> None:
 
   cls.__init__ = new_init  # type: ignore
   cls.__setattr__ = new_setattr  # type: ignore
+  setattr(cls, frozen_after_init.sentinel_attr, True) # type: ignore
```

Comment:
Remind me please why this change is needed? I remember a PR related to `sentinel_attr` but don't recall if it got merged.

Reply:
This is needed in order to then check whether a class was marked with `@frozen_after_init`. It is greatly streamlined in #8496!